src/displayapp/screens/Dice.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "components/motion/MotionController.h"
+#include "components/motor/MotorController.h"
+#include "displayapp/screens/Screen.h"
+#include "displayapp/widgets/Counter.h"
+#include "systemtask/SystemTask.h"
+
+#include <lvgl/lvgl.h>
+
+#include <array>
+#include <random>
+
+namespace Pinetime::Applications::Screens {
+  class Dice : public Screen {
+  public:
+    Dice(DisplayApp* app, Controllers::MotionController& motion, Controllers::MotorController& motor);
+    ~Dice() override;
+    void Roll();
+
+  private:
+    lv_obj_t* btnRoll;
+    lv_obj_t* btnRollLabel;
+    lv_obj_t* resultTotalLabel;
+    lv_obj_t* resultIndividualLabel;
+
+    std::mt19937 gen;
```

Comment:
We already use `rand()` to generate random numbers in Paddle, Twos and PTS watch face. Is there any reason to use another random number generator here?

Reply:
Sorry for the late reply! I noticed that, and so I did initially use `rand()` here as well in my earlier version. However, there are significant flaws with `rand()` and a few reasons that the functions used here instead (from the [C++11 `<random>` library](https://en.cppreference.com/w/cpp/header/random)) provide superior randomness. For the other apps mentioned, I think it doesn't matter too much to have great randomness, because the experience is fine as long as it _feels_ random enough for the user (and having fewer lines of code is better).

But for dice, where the whole _point_ of the app is randomness, it is important to have very fair randomness (especially if the dice were being used for a game or competition).

More details below:

(1) `std::rand()` is considered somewhat deprecated for nontrivial needs and does not guarantee quality of the randomness of the sequence produced. More details [here](https://en.cppreference.com/w/cpp/numeric/random/rand):

> There are no guarantees as to the quality of the random sequence produced. In the past, some implementations of rand() have had serious shortcomings in the randomness, distribution and period of the sequence produced (in one well-known example, the low-order bit simply alternated between 1 and 0 between calls). rand() is not recommended for serious random-number generation needs. It is recommended to use C++11's random number generation facilities to replace rand(). (since C++11)

(2) Using `rand() % x` opens the possibility for modulo bias: that is, if `RAND_MAX` is not divisible by `x`, then the last "segment" is shorter and generating smaller numbers is therefore more likely. With `RAND_MAX` being usually `2^32`, this bias is exceedingly slight, but again, it is best to correct for it when the randomness is the point of the app in question. While you can correct for this bias with a few extra lines of code with `rand()`, I think it is more elegant to use a [standard library function](https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution) that already does so by picking integers uniformly distributed on a specified closed interval.

(3) The `<random>` library here has the neat [`std::seed_seq` function](https://en.cppreference.com/w/cpp/numeric/random/seed_seq), which mixes multiple sources of randomness together (as opposed to `srand`, which takes in just 1 seed), and makes seeds distributed over the whole 32-bit range even when the inputs are close or poorly distributed. This app mixes the system tick count and the X/Y/Z components of the PineTime's sensitive motion controller (credit to @w4tsn #1199 for the motion controller idea!), making seed collision extremely unlikely.